### PR TITLE
Update DateRangeTP in [ Bind with TryParse ]

### LINKED ISSUE #27720: Update DateRangeTP in [ Bind with TryParse ]
- The use of `DateRangeTP` class in document and example is wrong(<https://learn.microsoft.com/en-us/aspnet/core/mvc/models/model-binding#bind-with-tryparse>)
- code example also need to be updated(<https://github.com/dotnet/AspNetCore.Docs/tree/main/aspnetcore/mvc/controllers/bind-tryparse/7.0-samples/BindUsingTryParse/BindTryParseMVC>)

#### document

the following paragraph need to be update to something like:

```diff
public class DateRangeTP
{
    public DateOnly? From { get; }
    public DateOnly? To { get; }

    public DateRangeTP(string from, string to)
    {
        if (string.IsNullOrEmpty(from))
            throw new ArgumentNullException(nameof(from));
        if (string.IsNullOrEmpty(to))
            throw new ArgumentNullException(nameof(to));

        From = DateOnly.Parse(from);
        To = DateOnly.Parse(to);
    }

    public static bool TryParse(string? value, out DateRangeTP? result)
    {
+      var range = value?.Split(',', StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
+      if (!range?.Length == 2)
-       if (string.IsNullOrEmpty(value) || value.Split('-').Length != 2)
        {
            result = default;
            return false;
        }

-       var range = value.Split(',');
        result = new DateRangeTP(range[0], range[1]);
        return true;
    }
}
```

```diff
+ The following controller action uses the DateRangeTP class to bind a date range:
- The following controller action uses the DateRDateRangeTPange class to bind a date range:
```

```diff
// GET /WeatherForecast/ByRangeTP?range=7/24/2022,07/26/2022
+ public IActionResult ByRangeTP([FromQuery] DateRangeTP range)
- public IActionResult ByRangeTP([FromQuery] DateRange range)
{
```

#### code example

- call method: <https://github.com/dotnet/AspNetCore.Docs/blob/0484277eef60875c7024781814d1f0111d72636b/aspnetcore/mvc/controllers/bind-tryparse/7.0-samples/BindUsingTryParse/BindTryParseMVC/Controllers/WeatherForecastController.cs#L66>
- DateRangeTP class:<https://github.com/dotnet/AspNetCore.Docs/blob/0484277eef60875c7024781814d1f0111d72636b/aspnetcore/mvc/controllers/bind-tryparse/7.0-samples/BindUsingTryParse/BindTryParseMVC/Models/DateRangeTP.cs#L22>

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: e24f48f8-6c4d-ecda-6ce6-a7826d396e99
* Version Independent ID: 9c4af557-8d06-4c6c-ad2a-0dd234547563
* Content: [Model Binding in ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/mvc/models/model-binding?view=aspnetcore-7.0#itp7)
* Content Source: [aspnetcore/mvc/models/model-binding.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/mvc/models/model-binding.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-mvc**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by BlueHtml**
Fixes #27720

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/AspNetCore.Docs?pullRequest=27738) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/AspNetCore.Docs?pullRequest=27738) before we can accept your contribution.<br/><br/>:x: BlueHtml [sign now](https://cla.dotnetfoundation.org/dotnet/AspNetCore.Docs?pullRequest=27738)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/AspNetCore.Docs?pullRequest=27738) it.</sub>

File: aspnetcore/mvc/controllers/bind-tryparse/7.0-samples/BindUsingTryParse/BindTryParseMVC/Models/DateRangeTP.cs [hunk 1]
```diff
@@ -19,13 +19,13 @@ public DateRangeTP(string from, string to)
 
     public static bool TryParse(string? value, out DateRangeTP? result)
     {
-        if (string.IsNullOrEmpty(value) || value.Split('-').Length != 2)
+        var range = value?.Split(',', StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
+        if (!range?.Length == 2)
```
Comment (fiyazbinhasan) [inline aspnetcore/mvc/controllers/bind-tryparse/7.0-samples/BindUsingTryParse/BindTryParseMVC/Models/DateRangeTP.cs:23]: It is giving me a `Cannot apply operator '!' to operand of type 'System.Nullable<int>'`
does the following make more sense?

` if (range?.Length != 2)`

Comment (BlueHtml) [reply aspnetcore/mvc/controllers/bind-tryparse/7.0-samples/BindUsingTryParse/BindTryParseMVC/Models/DateRangeTP.cs]: @fiyazbinhasan I'm sorry, I didn't verify it before, but now it has been fixed. And my English is not good, please forgive me.

Comment (fiyazbinhasan) [reply aspnetcore/mvc/controllers/bind-tryparse/7.0-samples/BindUsingTryParse/BindTryParseMVC/Models/DateRangeTP.cs]: no worries...very nice work 🐱 
@Rick-Anderson could you please review and merge

**Review by Rick-Anderson — APPROVED** [review]

**Comment by Rick-Anderson** [comment]
@BlueHtml nice work, keep the PRs coming

@fiyazbinhasan  thanks for your help

**Pull request merged**